stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala

```diff
@@ -1405,13 +1405,14 @@ private[stream] object Collect {
     new GraphStageLogic(shape) with InHandler with OutHandler {
       override def toString = s"MapAsyncUnordered.Logic(inFlight=$inFlight, buffer=$buffer)"
 
-      val decider =
+      private lazy val decider =
```

Comment:
Could you keep the lazy val change separate? I'm sure lazy val is probably ok here but lazy vals are converted into `synchronized` code when compiled to Java byte code.

Reply:
Ok, but I think that can be aline with MapAsync too.